src/hugr/views/sibling.rs

```diff
@@ -0,0 +1,572 @@
+            .expect("invalid subgraph")
+    }
+
+    /// Construct a [`SimpleReplacement`] to replace `self` with `replacement`.
+    ///
+    /// `replacement` must be a hugr with DFG root and its signature must
+    /// match the signature of the subgraph.
+    ///
+    /// We currently do not support inputs of the replacement graph being
+    /// copied.
+    ///
+    /// May return one of the following five errors
+    ///  - [`InvalidReplacement::InvalidDataflowGraph`]: the replacement
+    ///    graph is not a [`crate::ops::OpTag::DataflowParent`]-rooted graph,
+    ///  - [`InvalidReplacement::InvalidDataflowParent`]: the replacement does
+    ///    not have an input and output node,
+    ///  - [`InvalidReplacement::InvalidSignature`]: the signature of the
+    ///    replacement DFG does not match the subgraph signature,
+    ///  - [`InvalidReplacement::NonConvexSubgrah`]: the sibling subgraph is not
+    ///    convex, or
+    ///  - [`InvalidReplacement::InvalidCopy`]: the replacement has a copy at
+    ///    the input boundary.
+    pub fn create_simple_replacement(
+        &self,
+        replacement: Hugr,
+    ) -> Result<SimpleReplacement, InvalidReplacement>
+    where
+        Base: HugrView,
+    {
+        let removal = self.nodes().iter().copied().collect();
+
+        let rep_root = replacement.root();
+        let dfg_optype = replacement.get_optype(rep_root);
+        if dfg_optype.tag() != OpTag::Dfg {
+            return Err(InvalidReplacement::InvalidDataflowGraph);
+        }
+        let Some((rep_input, rep_output)) = replacement
+            .children(rep_root)
+            .take(2)
+            .collect_tuple()
+        else { return Err(InvalidReplacement::InvalidDataflowParent) };
+        if dfg_optype.signature() != self.signature() {
+            return Err(InvalidReplacement::InvalidSignature);
+        }
+        let rep_inputs = replacement
+            .node_outputs(rep_input)
+            .filter(|&p| {
+                replacement
+                    .get_optype(rep_input)
+                    .signature()
+                    .get(p)
+                    .is_some()
+            })
+            .map(|p| {
+                replacement
+                    .linked_ports(rep_input, p)
+                    .exactly_one()
+                    .map_err(|_| InvalidReplacement::InvalidCopy)
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        let rep_outputs = replacement.node_inputs(rep_output).filter(|&p| {
+            replacement
+                .get_optype(rep_output)
+                .signature()
+                .get(p)
+                .is_some()
+        });
+        let incoming = self.incoming_ports();
+        let outgoing = self
+            .outgoing_ports()
+            .map(|(n, p)| self.base.linked_ports(n, p));
+        let nu_inp = rep_inputs.into_iter().zip_eq(incoming).collect();
+        let nu_out = outgoing
+            .zip_eq(rep_outputs)
```

Comment:
I thought about this for awhile and agreed it was right :). However, it might be clearer if we reordered:
```
let nu_out = self
            .outgoing_ports()
            .zip_eq(rep_outputs)
            .flat_map(|((srcn, srcp), new_src)| {
                self.base
                    .linked_ports(srcn, srcp)
                    .map(move |target| (target, new_src))
            })
            .collect();
```
which although longer has one fewer `map` and I thought was a bit easier to follow. YMMV!

Reply:
Thanks, went with your suggestion with a few variable name changes. Hope this is clearer!